### DISCUSSION TITLE
can't get `short_url_length` within `api.configuration()`

### DISCUSSION THREAD

**Discussion by reagle** [tweepy/tweepy#2028] [Questions] [answered]
Until today I've been able to truncate my long tweets into 280 characters by assuming URLs are shortened to length of 23. Today, that's not working, and I thought perhaps the short url length had changed. It's also, more likely, because I'm using an emoji (see #1538), but I wanted to check and I see recommendations to check the [Twitter API configuration](https://www.geeksforgeeks.org/python-api-configuration-in-tweepy/), but that doesn't work.

What there a way to get API configuration values? Where did it go?

**Comment by Harmon758** [comment, marked as answer]
`API.configuration` was removed in [Tweepy v4.0.0](https://tweepy.readthedocs.io/en/v4.12.1/changelog.html#version-4-0-0-2021-09-25), as the Twitter API endpoint that it used was retired (see #1614).

Instead, if you're using Twitter API v1.1, you can use the indices field of the corresponding [URL object](https://developer.twitter.com/en/docs/twitter-api/v1/data-dictionary/object-model/entities#urls) to determine the length of the URL in the text of the Tweet. If you have the [Tweet object](https://developer.twitter.com/en/docs/twitter-api/v1/data-dictionary/object-model/tweet), you can retrieve its URL objects through the `entities` field and the `urls` field of that [Entities object](https://developer.twitter.com/en/docs/twitter-api/v1/data-dictionary/object-model/entities).

**Reply by reagle** [reply]
I'm concerned with `short_url_length` in composing tweets, so there is not Tweet object yet.

**Reply by Harmon758** [reply]
In that case, assuming URLs to be 23 characters should be fine.
See [the Twitter API documentation being referenced](https://developer.twitter.com/en/docs/counting-characters) in the issue that you linked.

**Reply by reagle** [reply]
Yes, I had to create a [little function](https://github.com/reagle/thunderdell/blob/main/utils/web.py#L132).

```
def len_twitter(text: str) -> int:
    """
    Twitter counts code units not code points as part of its
    character limit.
    https://developer.twitter.com/en/docs/counting-characters
    """

    return len(text.encode("utf-16-le")) // 2
```
